### PR TITLE
Initial controller commit

### PR DESCRIPTION
**Pull request by couralex6**
Adds:
* A new directory structure for the controller.
* A README inside the controller directory with installation
instructions.
* App.py containing REST API endpoint

Signed-off-by: Alex Courouble <acourouble@vmware.com>

### CODE REVIEW

File: controller/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+```cd chargepoint```
+
+### Start flask app
+
+```python -m flask run --host=0.0.0.0```
+
+### Configure your system
+
+1) Edit config.yaml to specify your ChargePoint station group and staion IDs. Whether you would like to stream load data, where to save files, enable DEBUG logging etc.
```
Comment (mkbhanda) [inline controller/README.md]: wondering if even station group and station-ids should just be environment variables. nit.
Also might need only station group if we support discovery, when we think ChargePoint station groups reporting into a controller.

File: controller/ev/chargepoint/app.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    Get current load
+    Note id will be parsed to extract a group of stations if specified
+    or a station, or a port on a station
+    """
+    cp.get_load(id)
+    return "Get load called on: " + id
+
+
+@app.route("/api/v1/curtail/<id>", methods=['POST'])
+def curtail(id):
```
Comment (mkbhanda) [inline controller/ev/chargepoint/app.py]: We have a method Curtail for now .. but something to think about .. Cody's use of positive and negative amounts, the later to ask a battery to deliver power to grid

File: controller/ev/chargepoint/app.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    """
+    Curtail a port given ID in URL and curtail type/amount in POST body.
+    Note id will be parsed to extract a group of stations if specified
+    or a station, or a port on a station
+    {
+        percentage: None,
+        absolute: 50
```
Comment (mkbhanda) [inline controller/ev/chargepoint/app.py]: Alex, please add time_interval as another form variable

and edit
cp.shed_load(id, absolute_amount = absolute_amount, time_interval= time_interval)

time_interval defaults to 0 to indicate stay curtailed till a clear command arrives.

Comment (mkbhanda) [reply controller/ev/chargepoint/app.py]: And I can make this edit when merged too, no problem.

File: controller/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+```python3 -m venv venv```
+
+### Start environment
+
+```. venv/bin/activate```
+
+### Install dependencies
+
+```pip install -r requirements.txt```
```
Comment (EnergySRE) [inline controller/README.md]: Might want to pack these steps into a Make target. Maybe `make env` or some such.

File: controller/ev/chargepoint/app.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
+from flask import Flask, request
```
Comment (EnergySRE) [inline controller/ev/chargepoint/app.py]: It'd be nice to have a module comment, to explain what to expect in this file.

File: controller/ev/chargepoint/app.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+@app.route("/api/v1/load/<id>")
+def get_load(id):
+    """
+    Get current load
+    Note id will be parsed to extract a group of stations if specified
+    or a station, or a port on a station
+    """
+    cp.get_load(id)
+    return "Get load called on: " + id
```
Comment (EnergySRE) [inline controller/ev/chargepoint/app.py]: Don't need to actually pass the result back to the caller?

**Pull request merged**